v2/tools/generator/internal/codegen/pipeline/repair_skipping_properties.go

```diff
@@ -184,46 +201,87 @@ func (detector *skippingPropertyDetector) findChains() *astmodel.PropertyReferen
 }
 
 // hasLinkFrom returns true if we already have a link from the specified property
-func (detector *skippingPropertyDetector) hasLinkFrom(ref astmodel.PropertyReference) bool {
+func (detector *skippingPropertyRepairer) hasLinkFrom(ref astmodel.PropertyReference) bool {
 	_, found := detector.links[ref]
 	return found
 }
 
-// checkChain checks for a gap in the specified property chain.
+// repairChain checks for a gap in the specified property chain.
 // start is the first property reference in the chain
-func (detector *skippingPropertyDetector) checkChain(start astmodel.PropertyReference) error {
+func (detector *skippingPropertyRepairer) repairChain(
+	start astmodel.PropertyReference,
+) (astmodel.TypeDefinitionSet, error) {
 	lastObserved, firstMissing := detector.findBreak(start, detector.wasPropertyObserved)
 	if firstMissing.IsEmpty() {
 		// Property was never discontinued
-		return nil
+		return nil, nil
 	}
 
-	_, reintroduced := detector.findBreak(firstMissing, detector.wasPropertyObserved)
+	lastMissing, reintroduced := detector.findBreak(firstMissing, detector.wasPropertyObserved)
 	if reintroduced.IsEmpty() {
 		// Property was never reintroduced
-		return nil
+		return nil, nil
 	}
 
 	// If the properties have the same type, we don't have a break here - so we check the remainder of the chain
 	// (This is Ok because the value serialized into the property bag from lastObserved will deserialize into the
 	// reintroduced property intact.)
 	typesSame, err := detector.propertiesHaveSameType(lastObserved, reintroduced)
 	if err != nil {
-		return errors.Wrapf(err, "failed to determine if properties %s and %s have the same type", lastObserved, reintroduced)
+		return nil, errors.Wrapf(
+			err,
+			"failed to determine if properties %s and %s have the same type",
+			lastObserved,
+			reintroduced)
 	}
 	if typesSame {
-		return detector.checkChain(reintroduced)
+		return detector.repairChain(reintroduced)
+	}
+
+	// We've found a skipping property with a different shape. If it's a TypeName we need to repair it.
+	// We do this by creating a new type that has the same shape as the missing property, injected just prior to
+	// reintroduction.
+
+	lastObservedPropertyType, ok := detector.lookupPropertyType(lastObserved)
+	if !ok {
+		// Should never fail, given the way findBreak() works
+		panic(fmt.Sprintf("failed to find type for property %s", lastObserved))
+	}
+
+	tn, ok := astmodel.AsInternalTypeName(lastObservedPropertyType)
+	if !ok {
+		// If not a type name, defer to our existing property conversion logic
+		// Continue checking the rest of the chain
+		return detector.repairChain(reintroduced)
```

Comment:
Won't this call the logic even though `!typesSame`?

Is that desired? I assume that this could/would happen in cases where the removed property was say, `*int`, and then it was added back as `*float`.

Do we already deal with primitive types here?